### PR TITLE
fix: correctly handle MDX layouts

### PR DESCRIPTION
**Pull request by danielvaijk**
# What is it?

- [ ] Feature / enhancement
- [x] Bug
- [ ] Docs / tests / types / typos
- [ ] Infra

# Description

This PR has fixes for two things:

1. I noticed that the link scripts in the `package.json` were broken since the relative path was incorrect.
2. A fix for https://github.com/QwikDev/qwik/issues/6844.

# Checklist:

- [x] My code follows the [developer guidelines of this project](https://github.com/QwikDev/qwik/blob/main/CONTRIBUTING.md)
- [x] I have performed a self-review of my own code
- [x] I have ran `pnpm change` and documented my changes
- [ ] I have made corresponding changes to the Qwik docs
- [x] Added new tests to cover the fix / functionality

### CODE REVIEW

File: packages/qwik-city/src/buildtime/markdown/mdx.ts [hunk 1]
```diff
@@ -79,10 +79,14 @@ export async function createMdxTransformer(ctx: BuildContext): Promise<MdxTransf
...
       const newDefault = `
-const WrappedMdxContent = () => {
-  return _jsxC(RenderOnce, {children: _jsxC(_createMdxContent, {}, 3, null)}, 3, ${JSON.stringify(key)});
+export default function MDXContent() {
+  const key = ${JSON.stringify(key)};
+  const content = _jsxC(_createMdxContent, {}, 3, null);
+
+  return typeof MDXLayout === "undefined" ?
+    _jsxC(RenderOnce, { children: content }, 3, key) :
+    _jsxC(RenderOnce, { children: _jsxC(MDXLayout, { children: content }, 3, key)});
```
Comment (wmertens) [inline packages/qwik-city/src/buildtime/markdown/mdx.ts:88]:
```js
  const children = typeof MDXLayout === "undefined" ? content : _jsxC(MDXLayout, { children: content }, 3)
  return _jsxC(RenderOnce, children, 3, key);
```

**Review by wmertens — CHANGES_REQUESTED** [review]
Nice but the return value is wrong I think, it needs to have the key on the top level component, no? And the static subtree flag should apply to all

File: packages/qwik-city/src/buildtime/markdown/mdx.ts (hunk 1 above)
Comment (danielvaijk) [reply packages/qwik-city/src/buildtime/markdown/mdx.ts]: I'm not 100% sure tbh. I do know the previous revision worked and so does the one with your suggestion, so I'll take the latter due to my lack of understanding of the rendering nuances of `_jsxC`. Thanks!

**Comment by wmertens** [comment]
@danielvaijk the E2E tests are failing, perhaps you can run them locally to find out the problem?

**Review by wmertens — DISMISSED** [review]
I fixed the E2E tests by no longer using internal API, which also helps with v2.

LGTM!

LGTM

**Review by wmertens — DISMISSED** [review]

**Review by wmertens — DISMISSED** [review]

**Review by wmertens — APPROVED** [review]

**Pull request merged**